### PR TITLE
Add default labels to solution operator deployment & pod

### LINKED ISSUE #2366: Solutions: Add the current Solution version in Operator & UI deployment
**Component**: salt, solutions

**Why this is needed**:
To be able to check the current version of Solution Operator & UI

**What should be done**:
Add a label (`app.kubernetes.io/version` ?) on both deployment with the Solution version as the value

**Implementation proposal** (strongly recommended):

**Test plan**:

### PR DESCRIPTION
**Pull request by alexandre-allard**
**Component**: salt, solutions

**Context**: 

**Summary**:

**Acceptance criteria**: 

---

Closes: #2366

### CODE REVIEW

File: salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml [hunk 1]
```diff
@@ -3,15 +3,29 @@ kind: Deployment
...
+    app.kubernetes.io/version: {{ version }}
+    app.kubernetes.io/component: operator
+    app.kubernetes.io/part-of: {{ solution }}
+    app.kubernetes.io/managed-by: salt
 spec:
   replicas: 1
   selector:
     matchLabels:
-      name: {{ solution }}-operator
+      app: {{ solution }}-operator
```
Comment (Ebaneck) [inline salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml:17]: I think this change here will break the Solutions test.
It is related to this line and must be fixed
`        Then we have 1 running pod labeled 'name=example-solution-operator' in namespace 'example-environment'
`

File: salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml [hunk 2]
```diff
@@ -38,4 +52,4 @@ spec:
       volumes:
         - name: operator-config
           configMap:
-            name: {{ solution }}-operator-config
\ No newline at end of file
+            name: {{ solution }}-operator-config
```
Comment (Ebaneck) [inline salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml]: What changed here??

**Review by Ebaneck — DISMISSED** [review]

File: salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml (hunk 2 above)
Comment (alexandre-allard) [reply salt/metalk8s/orchestrate/solutions/files/operator/deployment.yaml]: New line at the end of the file

**Review by gdemonet — CHANGES_REQUESTED** [review]
Could you also add `heritage` (equivalent of `managed-by`, often used by Helm)? Basically, make sure that both Operator and UI deployments use the same labelling principles (not just the *Deployment*s, all objects can/should be labelled).

It could be a feature of our `metalk8s_kubernetes` module to enforce those labels are always set, like it does for automatically adding the `metalk8s.scality.com/version` label... WDYT @TeddyAndrieux @NicolasT?

File: salt/_modules/metalk8s_kubernetes.py [hunk 3]
```diff
@@ -139,6 +139,9 @@ def method(manifest=None, name=None, kind=None, apiVersion=None,
             manifest.setdefault('metadata', {}).setdefault('labels', {})[
                 'metalk8s.scality.com/version'
             ] = match.group('version') if match else "unknown"
+            manifest['metadata']['labels']['app.kubernetes.io/managed-by'] = \
+                'salt'
+            manifest['metadata']['labels']['heritage'] = 'salt'
```
Comment (gdemonet) [inline salt/_modules/metalk8s_kubernetes.py:144]: :+1:

**Review by gdemonet — APPROVED** [review]
LGTM!

File: salt/_modules/metalk8s_kubernetes.py (hunk 3 above)
Comment (TeddyAndrieux) [reply salt/_modules/metalk8s_kubernetes.py:144]: Right now for things like coredns we put "metalk8s" as heritage

Edit: And for ingress we use "metalk8s" as managed-by also .... anyway we should have the same thing everywhere
Heritage = metalk8s
Managed-by = salt ?

Comment (TeddyAndrieux) [reply salt/_modules/metalk8s_kubernetes.py:144]: Also I think we shouldn't override these value if they are already in the manifest (and indeed it's the same for version just above)

**Pull request merged**